src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -1026,6 +1040,180 @@ end finite_measure --namespace
+            (ennreal.continuous_sub_left ennreal.one_ne_top).continuous_at,
+  exact ennreal.antitone_const_sub h,
+end
+
+lemma limsup_measure_le_of_le_liminf_measure_compl
+  {ι : Type*} {L : filter ι} {μ : measure α} {μs : ι → measure α}
+  [is_probability_measure μ] [∀ i, is_probability_measure (μs i)]
+  {E : set α} (E_mble : measurable_set E) (h : μ Eᶜ ≤ L.liminf (λ i, μs i Eᶜ)) :
+  L.limsup (λ i, μs i E) ≤ μ E :=
+compl_compl E ▸ (limsup_measure_compl_le_of_le_liminf_measure (measurable_set.compl E_mble) h)
+
+variables [topological_space α] [opens_measurable_space α]
+
+/-- One pair of implications of the portmanteau theorem:
+For a sequence of Borel probability measures, the following two are equivalent:
+
+(C) The limsup of the measures of any closed set is at most the measure of the closed set
+under a candidate limit measure.
+
+(O) The liminf of the measures of any open set is at least the measure of the open set
+under a candidate limit measure.
+-/
+lemma limsup_measure_closed_le_iff_liminf_measure_open_ge
+  {ι : Type*} {L : filter ι} {μ : measure α} {μs : ι → measure α}
+  [is_probability_measure μ] [∀ i, is_probability_measure (μs i)] :
+  (∀ F, is_closed F → L.limsup (λ i, μs i F) ≤ μ F)
+    ↔ (∀ G, is_open G → μ G ≤ L.liminf (λ i, μs i G)) :=
+begin
+  split,
+  { intros h G G_open,
+    exact le_measure_liminf_of_limsup_measure_compl_le
+          G_open.measurable_set (h Gᶜ (is_closed_compl_iff.mpr G_open)), },
+  { intros h F F_closed,
+    exact limsup_measure_le_of_le_liminf_measure_compl
+          F_closed.measurable_set (h Fᶜ (is_open_compl_iff.mpr F_closed)), },
+end
+
+end limsup_closed_le_and_le_liminf_open -- section
+
+section tendsto_of_null_frontier
+/-! ### Portmanteau: limit of measures of Borel sets whose boundary carries no mass in the limit
+
+In this section we prove that for a sequence of Borel probability measures on a topological space
+and its candidate limit measure, either of the following equivalent conditions:
+  (C) For any closed set `F` in `α` the limsup of the measures of `F` is at most the limit
+      measure of `F`
+  (O) For any open set `G` in `α` the liminf of the measures of `G` is at least the limit
+      measure of `G`
+implies that
+  (B) For any Borel set `E` in `α` whose boundary `∂E` carries no mass under the candidate limit
+      measure, we have that the limit of measures of `E` is the measure of `E` under the
+      candidate limit measure.
+-/
+
+variables {α : Type*} [measurable_space α]
+
+lemma tendsto_measure_of_le_liminf_measure_of_limsup_measure_le
+  {ι : Type*} {L : filter ι} {μ : measure α} {μs : ι → measure α}
+  {E₀ E E₁ : set α} (E₀_subset : E₀ ⊆ E) (subset_E₁ : E ⊆ E₁) (nulldiff : μ (E₁ \ E₀) = 0)
+  (h_E₀ : (μ : measure α) E₀ ≤ L.liminf (λ i, (μs i : measure α) E₀))
+  (h_E₁ : L.limsup (λ i, (μs i : measure α) E₁) ≤ (μ : measure α) E₁) :
+  L.tendsto (λ i, (μs i : measure α) E) (𝓝 (μ E)) :=
```

Comment:
you shouldn't need the coercions to measures here.

Reply:
There were apparently still many of these left, sorry! I went through the whole file and hopefully removed all unnecessary coercions to measures.